index.js

```diff
@@ -0,0 +1,133 @@
+const jsonValidator = require('json-dup-key-validator');
+const path = require('path');
+const fs = require('fs');
+const util = require('util');
+
+const { promisify } = util;
+const readDirAsync = promisify(fs.readdir);
+const readFileAsync = promisify(fs.readFile);
+
+const loadFilesNames = async directoryPath => await readDirAsync(directoryPath);
+
+const loadFilesContent = async (filesNames, dir) => {
+  const filesData = [];
+  const duplicatedKeys = [];
+
+  await Promise.all(
+    filesNames
+      .filter(fileName => fileName.endsWith('.json'))
+      .map(async fileName => {
+        const fileData = await readFileAsync(`./${dir}/${fileName}`, 'utf8');
+        const parsedFileData = await JSON.parse(fileData);
+        filesData.push({
+          fileName,
+          fileData: parsedFileData
+        });
+
+        const duplicatedKey = await jsonValidator.validate(fileData, false);
+        if (duplicatedKey) {
+          duplicatedKeys.push({
+            type: '[DUPLICATED]',
+            key: duplicatedKey,
+            fileName
+          });
+        }
+      })
+  );
+
+  return [filesData, duplicatedKeys];
+};
+
+const getAllFilesData = async dir => {
+  const directoryPath = path.join(__dirname, dir);
+  const filesNames = await loadFilesNames(directoryPath);
+  return await loadFilesContent(filesNames, dir);
+};
+
+const createKeysList = filesData => {
+  return filesData
+    .map(data => {
+      return Object.keys(data.fileData);
+    })
+    .reduce((acc, curr) => [...acc, ...curr], []);
+};
+
+const countKeysInFiles = keysList => {
+  const countedKeys = keysList.reduce((prev, cur) => {
```

Comment:
Consider removal of the code block and just let the arrow implicitly return the `keysList.reduce` result.